File: LeetCode/Java Solutions/Sum of Left Leaves/sum.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int sumOfLeftLeaves(TreeNode root) {
        if (root == null) return 0;
        
        int leftSum = 0;
        if (root.left != null && isLeaf(root.left)) {
            leftSum = root.left.val;
        }
        
        return leftSum + sumOfLeftLeaves(root.left) + sumOfLeftLeaves(root.right);
    }
    
    private boolean isLeaf(TreeNode node) {
        return node != null && node.left == null && node.right == null;
    }
}

File: LeetCode/Python Solutions/Sum of Left Leaves/sum.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumOfLeftLeaves(self, root: TreeNode) -> int:
        if not root:
            return 0
        
        def isLeaf(node):
            return node is not None and node.left is None and node.right is None

        def dfs(node):
            if not node:
                return 0
            
            left_sum = 0
            if node.left and isLeaf(node.left):
                left_sum = node.left.val
            return left_sum + dfs(node.left) + dfs(node.right)
        
        return dfs(root)